openpype/hosts/blender/api/workio.py

```diff
@@ -87,3 +97,51 @@ def work_root(session: dict) -> str:
     if scene_dir:
         return str(Path(work_dir, scene_dir))
     return work_dir
+
+
+def check_workfile_up_to_date() -> bool:
+    """Check if the current workfile is out of date.
+
+    This is based on last modification date, so if a user modifies an out of
+    date workfile, this will return `False`. Also, in case of partial publish,
+    this will return `True`.
+
+    Returns:
+        bool: True if the current workfile is up to date.
+    """
+
+    project_name = get_current_project_name()
+    asset_name = get_current_asset_name()
+
+    # Get date and time of the latest published workfile
+    last_published_version = get_last_version_by_subset_name(
+        project_name,
+        next(
+            (
+                repre.get("context", {}).get("subset", "")
+                for repre in get_representations(
+                    project_name,
+                    context_filters={
+                        "asset": asset_name,
+                        "family": "workfile",
+                        "task": {"name": get_current_task_name()},
+                    }
+                )
+            ),
+            "",
+        ),
+        asset_name=asset_name,
+        fields=["data"]
+    )
+
+    # Check if version exists
+    if not last_published_version:
+        return True
+
+    scene = bpy.context.scene
+
+    last_published_time = last_published_version["data"]["time"]
+    if scene.get("op_published_time"):
+        return last_published_time <= scene["op_published_time"]
+    else:
+        return False
```

Comment:
Wouldn't it be better to return `True` in this case? I guess this happens when users start using this feature in existing projects, and they have published workfiles already. I think it would be confusing to suddently get a message that their workfile is not Up to date.

Reply:
On one hand yes, and I agree that it could be confusing. On the other hand, in this fallback state where the system can't determine if the workfile is up to date or not, I think it's safer to wrongly mark it as out of date, in which case the worst thing that could happen is downloading a workfile while it wasn't necessary, than to mark it as up to date, in which case we would risk publishing out of date workfile, which is all that this system is trying to prevent.

This is why, in my opinion, returning `False` is the way to go. In practice, as I think downloading last workfile is what should generally be recommended to artists, and there is no risk in doing it. Artists will most likely just click ok and not worry too much about it.

Reply:
I agree with Sharkitty. As this is a new feature, releasing a new OP version must come with information for artists about new behaviours they might encounter.  
Also, if we set it to `True`, we may leave issues to go through, like some one creates by mistake a new workfile from scratch (build or whatever), this workfile won't contain the key, and then he tries to publish. The user will have no any warning or anything to think about what is going on.

The fix could be to register a handler at the file opening to set this key with the appropriate value if it doesn't exist.